src/common/blkdev.cc

```diff
@@ -89,7 +89,10 @@ int get_block_device_base(const char *dev, char *out, size_t out_len)
     if (*p == '/')
       *p = '!';
 
-  snprintf(fn, sizeof(fn), "%s/sys/block/%s", sandbox_dir, devname);
+  if (static_cast<size_t>(snprintf(fn, sizeof(fn), "%s/sys/block/%s",
```

Comment:
This one confuses me. It looks like it shouldn't ever be true? What was the diagnostic issued?

Reply:
quote from manpage

> These functions return the number of characters printed (not including the trailing `\0' used to end output to strings), except for snprintf() and vsnprintf(), which return the number of characters that would have been printed if the size were unlimited (again, not including the final `\0').  These functions return a negative value if an error occurs.

the warning was the one issued from `-Wformat-truncation=`.

Reply:
Ah, I'm with you-- I forgot about the negative values. Good catch. :-)